android/app/src/main/java/org/uwaterloo/subletr/pages/home/map/HomeMapChildViewModel.kt

```diff
@@ -1,17 +1,68 @@
 package org.uwaterloo.subletr.pages.home.map
 
-import android.util.Log
+import android.location.Location
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.subjects.BehaviorSubject
 import kotlinx.coroutines.coroutineScope
+import org.uwaterloo.subletr.api.models.ListingSummary
+import org.uwaterloo.subletr.api.models.ResidenceType
+import org.uwaterloo.subletr.enums.Gender
+import org.uwaterloo.subletr.enums.HousingType
 import org.uwaterloo.subletr.infrastructure.SubletrChildViewModel
+import org.uwaterloo.subletr.pages.home.HomePageUiState
 import org.uwaterloo.subletr.services.ILocationService
+import java.time.OffsetDateTime
 import javax.inject.Inject
 
 class HomeMapChildViewModel @Inject constructor(
 	private val locationService: ILocationService,
 ): SubletrChildViewModel<HomeMapUiState>() {
-	suspend fun getLocation() {
-		coroutineScope {
-			Log.d("LocationLogs", locationService.getLocation().toString())
+	suspend fun getLocation(): Location? {
+		return coroutineScope {
+			return@coroutineScope locationService.getLocation()
 		}
 	}
+
+	val addressSearchStream: BehaviorSubject<String> = BehaviorSubject.createDefault("")
+	val timeToDestinationStream: BehaviorSubject<Float> = BehaviorSubject.createDefault(0f)
+	val transportationMethodStream: BehaviorSubject<HomeMapUiState.TransportationMethod> =
+		BehaviorSubject.createDefault(HomeMapUiState.TransportationMethod.ALL)
+
+	override val uiStateStream: Observable<HomeMapUiState> = Observable.combineLatest(
+		addressSearchStream,
+		timeToDestinationStream,
+		transportationMethodStream,
+	) { addressSearch, timeToDestination, transportationMethod ->
+		HomeMapUiState.Loaded(
+			addressSearch = addressSearch,
+			timeToDestination = timeToDestination,
+			transportationMethod = transportationMethod,
+			locationRange = HomePageUiState.LocationRange(null, null),
+			priceRange = HomePageUiState.PriceRange(null, null),
+			roomRange = HomePageUiState.RoomRange(null, null, null, false),
+			genderPreference = Gender.FEMALE,
+			houseTypePreference = HousingType.HOUSE,
+			listingItems = HomePageUiState.ListingItemsModel(
+				listings = listOf(
+					ListingSummary(
+						listingId = 1,
+						address = "123 University Ave.",
+						distanceMeters = 25f,
+						price = 100,
+						roomsAvailable = 2,
+						roomsTotal = 2,
+						bathroomsAvailable = 2,
+						bathroomsEnsuite = 0,
+						bathroomsTotal = 2,
+						leaseStart = OffsetDateTime.now(),
+						leaseEnd = OffsetDateTime.now(),
+						imgIds = emptyList(),
+						residenceType = ResidenceType.apartment,
+					)
```

Comment:
I see this one is hard coded in, what's the plan to get actual listings loaded in?

Reply:
We have to move the streams into the parent so that both the map view and list view share the same state.
This PR just makes the view for now and I'll have to follow up with a separate PR to move all the streams when the current PR around the home page are merged in.

Reply:
Yep, definitely best to move that to a separate PR. Resolved :)